cip/1.accepted/CIP2016-06-14-Define-comparability-and-equality-as-well-as-orderability-and-equivalence.adoc

```diff
@@ -157,19 +209,39 @@ We propose that <<comparability-con,comparability>> should be defined between an
 [[equality-def,equality]]
 ==== Equality ====
 
+Given that values are only comparable within their most specific type, and comparability and equality are tied together, equlity of values of different types always evaluates to `null`.
 In order to align equality with <<comparability-def>>, we change equality of lists and maps that contain `null` values to treat those values in the same way as if they would have been compared outside of those lists and maps, as individual, simple values.
 
 ===== List equality =====
 
 Specifically, we propose to redefine how equality works for lists in Cypher.
-To determine if two lists `l1` and `l2` are equal, we propose two simple tests, as exemplified by the following:
+The equality of two lists `a` and `b` is defined as the <<conjunction>> of `size(a) = size(b)` and a pairwise comparison of all elements of the list.
+If `a = b` evaluates to `null` these lists are <<incomparable>>.
 
-* `l1` and `l2` must have the same size, i.e. inversely `size(l1) <> size(l2) \=> l1 <> l2`
-* the pairwise elements of both `l1` and `l2` must be equal, i.e.
 ----
-[a1, a2, ..., an] = [b1, b2, ..., bn]
-<=>
-a1 = b1 && a2 = b2 && ... && an = bn
+    [1, 2] = [1]
+<=> size([1, 2]) = size([1]) AND 1 = 1 AND 2 = null
+<=> false                    AND true  AND null
+<=> false
+
+    [null] = [1]
+<=> size([null]) = size[1]) AND null = 1
+<=> true                    AND null
+<=> null
+
+    ["a"] = [1]
+<=> size(["a"]) = size[1]) AND "a" = 1
+<=> size(["a"]) = size[1]) AND "a" = 1
+<=> true                   AND null
+<=> null
+----
+
+The same logic applies recursively when comparing nested lists.
+----
+    [[1]] = [[1], [null]]
+<=> size([[1]]) = size([[1], [null]]) AND [1] = [1] AND null = [null]
```

Comment:
Just for the sake of the example, we could explain this with an additional step I think:
```suggestion
<=> size([[1]]) = size([[1], [null]]) AND [1] = [1]                       AND null = [null]
<=> 1 = 2                             AND size([1]) = size([1]) AND 1 = 1 AND null
```